### ISSUE TITLE
progress indicator

### ISSUE THREAD

**Issue by Luceos** [flagrow/upload#35]
> it would be very useful to add some kind of visual feedback (an animation, or a percentage of some sort) to clearly show when the file upload to the server is happening.
> This is even more relevant in case of large files uploads, when the user has no clear perception whether the upload is still ongoing or has stuck for some reason.

- http://tympanus.net/codrops/2013/12/12/progress-button-styles/

**Comment by webeindustry** [comment]
This will be very useful. If someone tries to post before the upload is complete it will make the post without the image. That could lead to confusion and support issues.

**Comment by yekenuk** [comment]
Vote for this too. Some users think the upload isn't working as they have no indication on mobile that it is doing something.

**Comment by Stef-Dan** [comment]
A progress bar would be perfect.

**Comment by noriods** [comment]
I'm happy to give this a shot.

@luceos how do you feel about disabling the "Post Discussion" button while the file uploads?

**Comment by luceos** [comment]
@noriods makes sense 👍 But let's keep it stupid simple first. Complexity can always added, it's harder to leave it out.
